packages/server/src/automations/loopUtils.ts

```diff
@@ -0,0 +1,136 @@
+import { AutomationErrors } from "../constants"
+import * as automationUtils from "./automationUtils"
+import env from "../environment"
+import { LoopInput } from "../definitions/automations"
+
+export function initializeTempOutput(loopSteps: any, iterationCount: number) {
+  return { items: loopSteps, iterations: iterationCount }
+}
+
+export function typecastLoopInput(
+  loopStep: any,
+  loopStepNumber: number,
+  step: any,
+  tempOutput: any,
+  updateContextAndOutput: Function
+): boolean {
+  try {
+    loopStep.inputs.binding = automationUtils.typecastForLooping(
+      loopStep.inputs as LoopInput
+    )
+  } catch (err) {
+    updateContextAndOutput(loopStepNumber, step, tempOutput, {
+      status: AutomationErrors.INCORRECT_TYPE,
+      success: false,
+    })
+    return false
+  }
+  return true
+}
+
+export function getLoopItems(binding: any, option: string | undefined): any[] {
```

Comment:
Would it be possible to provide better typing for some of this stuff than `any`? It looks like this function handles if the binding is either a string or an array, this could likely be typed with `string | any[]` even if we don't know what the array contains.